### PR TITLE
Add redirected Relu grad and test

### PR DESCRIPTION
**Pull request by ludwigschubert**
Adds relu gradient override, but does not yet make it the default for `render_vis`.
Also adds test to `gradient_override`.

### CODE REVIEW

File: lucid/misc/redirected_relu_grad.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+"""Redirected ReLu Gradient Overrides
+
+When visualizing models we often[0] have to optimize through ReLu activation
```
Comment (colah) [inline lucid/misc/redirected_relu_grad.py]: Being very nitpicky, I'd probably structure this a bit differently:

"When we visualize ReLU networks, the initial random input we give the model may not cause the neuron we're visualizing to fire at all. For a ReLU neuron, this means that no gradient flow backwards and the visualization never takes off. One solution would be to find the pre-ReLU tensor, but that can be tedious."

"These functions provide a more convenient solution: temporarily override the gradient of ReLUs to allow gradient to flow back through the ReLU -- even if it didn't activate and had a derivative of zero -- allowing the visualization process to get started."

File: lucid/misc/redirected_relu_grad.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+with gradient_override_map({'Relu': redirected_relu_grad}):
+  model.import_graph(…)
+```
+
+Discussion:
+ReLus block the flow of the gradient during backpropagation when their input is
+negative. ReLu6s also do so when the input is larger than 6. These overrides
+change this behavior to allow gradient pushing the input into a desired regime
+between these points.
```
Comment (colah) [inline lucid/misc/redirected_relu_grad.py]: Clarify that this only happens when the gradient would be zero if we didn't intervene?

File: lucid/misc/redirected_relu_grad.py [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+"""Redirected ReLu Gradient Overrides
+
+When visualizing models we often[0] have to optimize through ReLu activation
+functions. Where accessing pre-relu tensors is too hard, we use these
+overrides to allow gradient to flow back through the ReLu—even if it didn't
```
[the remark this answers is not in the document]
Comment (ludwigschubert) [reply lucid/misc/redirected_relu_grad.py]: That and my favorite ellipsis character…

File: lucid/misc/redirected_relu_grad.py (hunk 1 above)
Comment (ludwigschubert) [reply lucid/misc/redirected_relu_grad.py]: ACK

File: lucid/misc/redirected_relu_grad.py (hunk 2 above)
Comment (ludwigschubert) [reply lucid/misc/redirected_relu_grad.py]: ACK

**Pull request merged**